.circleci/RoboFile.php

```diff
@@ -42,6 +51,165 @@ class RoboFile extends Tasks {
    */
   protected $bltConfig;
 
+  /**
+   * Perform a release in github.
+   */
+  public function jobGithubRelease() {
+
+    $last_version = $this->getLastVersion();
+    // Increment the last version by 1.
+    $version = $this->incrementVersion($last_version);
+    $this->yell("Releasing $version");
+
+    // Get a list of all commits since the last version until now.
+    exec("git log --pretty=format:%h $last_version...HEAD", $commit_hashes);
+
+    $changes = [];
+    // Build an array of change strings.
+    foreach ($commit_hashes as $hash) {
+      exec("git log --format=%B -n 1 $hash", $log);
+      $log = implode(';', array_filter($log));
+      $changes[] = "* $log ($hash)";
+    }
+
+    // Set module and profile version. Then update the changelog.
+    $this->setVersions($version);
+    $this->updateChangelog($version, $changes);
+
+    // Commit all the changes and push to github.
+    $result = $this->taskGitStack()
+      ->add('-A')
+      ->commit("Release $version")
+      ->pull()
+      ->push()
+      ->run();
+
+    if (!$result->wasSuccessful()) {
+      throw new Exception('Release commit was unsuccessful');
+    }
+
+    // Create a new release in github. This will generate a tag which will be
+    // used in another CircleCI task.
+    $result = $this->taskGitHubRelease($version)
+      ->accessToken(getenv('GITHUB_TOKEN'))
+      ->uri('SU-HSDO/suhumsci')
+      ->description("Release $version")
+      ->changes($changes)
+      ->run();
+
+    if (!$result->wasSuccessful()) {
+      throw new Exception('Release was unsuccessful');
+    }
+  }
+
+  /**
+   * Get the last version from the profile.
+   *
+   * @return string
+   *   Last semver version.
+   */
+  protected function getLastVersion() {
```

Comment:
So this locks the build version with the profile version? Perhaps we need a way to determine what the build version is as I don't think these will be 1-1 on every project.

Reply:
I was grabbing the latest git tag, but i wasnt confident that was in semver

Reply:
We could change this logic if we use this on other projects

Reply:
Aye, sorry, I've been doing review through the ACSF lens. I'll re-adjust my thinking and take another pass.